api/controllers/projectGroupV2.js

```diff
@@ -0,0 +1,303 @@
+// Imports
+const defaultLog = require('winston').loggers.get('default');
+const Actions    = require('../helpers/actions');
+const projectDAO = require('../dao/projectDAO');
+const projectGroupDAO = require('../dao/projectGroupDAO');
+
+// Constants
+const PUBLIC_ROLES = ['public'];
+const SECURE_ROLES = ['sysadmin', 'staff'];
+// Vars
+/* put any needed local variables here */
+// functions
+/* put any needd local function here */
+
+// Exports
+
+// OPTIONS
+exports.projectGroupOptions = function (args, res, rest) 
+{
+  res.status(200).send();
+};
+
+exports.projectGroupOptionsProtected = function (args, res, rest) 
+{
+  res.status(200).send();
+};
+
+// ? no getGroups (list)
+// ? no getGroup (resource)
+// ? no getMemeber (resource)
+
+// POST (Protected only createGroup)
+exports.createGroup = async function (args, res, rest) 
+{
+    defaultLog.debug('>>> {POST} /Projects/{id}/Groups');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('projId') && args.swagger.params.hasOwnProperty('group'))
+        {
+            let projectId = args.swagger.params.projId.value;
+            let group = JSON.parse(args.swagger.params.group.value);
+
+            defaultLog.debug(' Creating group for project ' + projectId);
+            defaultLog.debug(' group: ' + JSON.stringify(group));
+
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
+
+            if(project)
+            {
+                let savedGroup = await projectGroupDAO.createGroup(args.swagger.params.auth_payload.preferred_username, group.name, project);
+                savedGroup = projectGroupDAO.groupHateoas(savedGroup, SECURE_ROLES);
+                return Actions.sendResponseV2(res, 201, savedGroup);
```

Comment:
Nice! Like the proper response codes.

Reply:
Thanks! Let me know if you spot any that should be corrected or improved. Copy pasting may be easy but leads to missed values...